sql/expression/literal.go

```diff
@@ -64,24 +65,28 @@ func (lit *Literal) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 }
 
 func (lit *Literal) String() string {
-	switch v := lit.value.(type) {
+	switch litVal := lit.value.(type) {
 	case int, int8, int16, int32, int64, uint, uint8, uint16, uint32, uint64:
-		return fmt.Sprintf("%d", v)
+		return fmt.Sprintf("%d", litVal)
 	case string:
 		switch lit.fieldType.Type() {
 		// utf8 charset cannot encode binary string
 		case query.Type_VARBINARY, query.Type_BINARY:
-			return fmt.Sprintf("'0x%X'", v)
+			return fmt.Sprintf("'0x%X'", litVal)
 		}
-		return fmt.Sprintf("'%s'", v)
+		// Conversion of \' to \'\' required as this string will be interpreted by the sql engine.
+		// Backslash chars also need to be replaced.
+		escaped := strings.ReplaceAll(litVal, "'", "''")
+		escaped = strings.ReplaceAll(escaped, "\\", "\\\\")
```

Comment:
do we need any more? `%`, `\n`, `\t`, `\v`, etc?

Reply:
Added with https://github.com/dolthub/go-mysql-server/pull/1639/commits/629fa869038993b78661aafa7ee87c46bef5152b